### ISSUE TITLE
Validity bug?

### ISSUE THREAD

**Issue by matteocorti** [matteocorti/check_ssl_cert#8]
I am trying to use this on a centos 7 machine with nagios 4 + it all
works fine but how do we get it to warn show crytical on the number of
days?  this appears not to work whatever we set the min number of days
too.

```
# ./check_ssl_cert -H www.cnn.com -w 1000
SSL_CERT OK - X.509 certificate for 'a.ssl.fastly.net' from 'DigiCert
SHA2 High Assurance Server CA' valid until Feb  6 12:00:00 2018 GMT
(expires in 707 days)|days=707;1000;;;

```

**Comment by matteocorti** [comment]
To me

**Comment by matteocorti** [comment]
The validity check is done with openssl regardless if the Date::Perl plugin is installed.

openssl checks if the certificate will expire in a certain amount of seconds:

From the documentation of x509

```
       -checkend arg
           checks if the certificate expires within the next arg seconds and exits non-zero if yes it will expire or zero if not.

```

In the [x509 implementation](https://github.com/openssl/openssl/blob/master/apps/x509.c) `checkend` is a signed int (max 2,147,483,647 seconds or 24'855 days). And x509 performs a validity check so it seems not to be an overflow problem

**Comment by matteocorti** [comment]
It seems that this is long living but in check_ssl. The expiration was checked only if --ignore-exp was specified ...

**Comment by matteocorti** [comment]
Fixed in ef4261bf3ed10a3072a13e1b7c72b413ffb16b7e
